src/reducers/AddEditSiteReducer.js

```diff
@@ -82,77 +118,87 @@ const siteIsReadyForUpload = ({latitudeText, longitudeText, siteTitleText, siteD
     return latitudeText && longitudeText && siteTitleText && siteDescriptionText && siteDirectionsText && siteNearestTownText && accessibilityOption && priceOption && siteDetailCheckboxesKeys.facilities.length > 0 && siteDetailCheckboxesKeys.features.length > 0;
 };
 
-export default (state = INITIAL_STATE, action) => {
+const formReducer = prefix => (state = INITIAL_STATE, action) => {
     const {type, payload} = action;
 
     switch (type) {
 
-        case LATITUDE_TEXT_UPDATED:
+        case NEW_SITE_TO_EDIT:
+            if (prefix === site_form_type.EDIT) {
+                const {siteToEdit} = payload;
+                const siteStateToDispatch = setInitialEditStateFromSite({siteToEdit});
+
+                return {...siteStateToDispatch};
+            } else {
+                return state;
+            }
+
+        case `${prefix}_${LATITUDE_TEXT_UPDATED}`:
             const {latitudeText} = payload;
             const cleanLatText = removeNonNumbers(latitudeText);
             const readyLatitude = _.toNumber(cleanLatText);
 
             return {...state, latitudeText: cleanLatText, readyLatitude};
 
-        case LONGITUDE_TEXT_UPDATED:
+        case `${prefix}_${LONGITUDE_TEXT_UPDATED}`:
             const {longitudeText} = payload;
             const cleanLongText = removeNonNumbers(longitudeText);
             const readyLongitude = _.toNumber(cleanLongText);
 
             return {...state, longitudeText: cleanLongText, readyLongitude};
 
-        case SITE_TITLE_TEXT_CHANGED:
+        case `${prefix}_${SITE_TITLE_TEXT_CHANGED}`:
```

Comment:
All of these take the prefix of either 'add' or 'edit' to make the types dynamic and reusable for two different forms of state.